server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -176,23 +176,52 @@ def to_representation(self, instance):
 
     def translate_filters(self, model, filters):
         """
-        Iterate through the filters and translate them to use our internal db representation.
+        Recursively translate mongo queries from our external API to our internal db representation.
+
+        Mongo queries follow one of the following forms:
+            1. {"key": "value"} - ex. {"id": "thing1"}
+            2. {"key": [list of items] - ex. {'$in': ['this', 'or_this']
+            2. {"key": [list of queries] - ex. {'$and': [{query_1}, {query_2}]
+            3. {"$key": {dict}} - ex. {"$elemMatch": {query}}
+            4. {"key": {dict}} - ex. {"id": {"$exists": 1}}
+
+        For case 1, the simple case, the key may need to be translated. Case 2 and 3, the key does
+        not need to be translated because it is a mongo operator, but there may be nested queries
+        that will need to be translated. Case 4 requires translation of the key and then recursive
+        translation of the value.
 
         :param model: the class that defines this document's fields
         :type  model: sublcass of mongoengine.Document
         :param filters: dict used to filter results
-        :type  filters: dict, following mongo syntax
+        :type  filters: dict, following mongo query syntax
 
-        :return: the same filters dictionary, with fields translated to the new db representation
-        :rtype:  dict, following mongo syntax
+        :return: filters dictionary, with fields translated
+        :rtype:  dict, following mongo query syntax
         """
         translated_dict = {}
         for key, value in filters.iteritems():
-            if key in self._remapped_fields.itervalues():
+            if key.startswith('$'):
+                if type(value) is list:
```

Comment:
This style of type checking is, in my opinion, overly rigid. At the very least, it should be `isinstance(value, list)`.